src/Raven.Client/Documents/Commands/Batches/PutAttachmentCommandData.cs

```diff
@@ -0,0 +1,44 @@
+﻿using System;
+using System.IO;
+using Raven.Client.Documents.Conventions;
+using Sparrow.Json;
+using Sparrow.Json.Parsing;
+
+namespace Raven.Client.Documents.Commands.Batches
+{
+    public class PutAttachmentCommandData : ICommandData
+    {
+        public PutAttachmentCommandData(string documentId, string name, Stream stream, string contentType, long? etag)
+        {
+            if (string.IsNullOrWhiteSpace(documentId))
+                throw new ArgumentNullException(nameof(documentId));
+            if (string.IsNullOrWhiteSpace(name))
+                throw new ArgumentNullException(nameof(name));
+
+            Key = documentId;
+            Name = name;
+            Stream = stream;
```

Comment:
We need to validate that:
* The stream is seekable.
* That the position is 0.

Need to explain that we require this to support failover scenarios, where we might sent the attachment to more than a single location.